Detour/Source/DetourNavMeshQuery.cpp

```diff
@@ -1916,7 +1918,9 @@ dtStatus dtNavMeshQuery::findStraightPath(const float* startPos, const float* en
 		}
 	}
 
-	stat = appendVertex(closestEndPos, DT_STRAIGHTPATH_END, 0,
+	dtStraightPathFlags endflag;
+	endflag.straightpathEnd = true;
```

Comment:
This is clumsy. Perhaps a helper using the 'builder' pattern would tidy it up, allowing you to write e.g. `straightPathFlags().straightpathEnd(true)`. Or perhaps static constants for the values would be useful.